ui/selectors/approvals.ts

```diff
@@ -1,19 +1,78 @@
 import { ApprovalControllerState } from '@metamask/approval-controller';
 import { ApprovalType } from '@metamask/controller-utils';
+import { transactionMatchesNetwork } from '../../shared/modules/transaction.utils';
+import { TransactionMeta } from '../../shared/constants/transaction';
+import { getCurrentChainId } from './selectors';
 
 type ApprovalsMetaMaskState = {
   metamask: {
     pendingApprovals: ApprovalControllerState['pendingApprovals'];
+    unapprovedTxs: {
+      [transactionId: string]: TransactionMeta;
+    };
   };
 };
 
-export function hasPendingApprovalsSelector(
+const getApprovalRequestsByType = (
   state: ApprovalsMetaMaskState,
   approvalType: ApprovalType,
-) {
+) => {
   const pendingApprovalRequests = Object.values(
     state.metamask.pendingApprovals,
   ).filter(({ type }) => type === approvalType);
 
+  return pendingApprovalRequests;
+};
+
+const hasUnapprovedTransactionsInCurrentNetwork = (
+  state: ApprovalsMetaMaskState,
+) => {
+  const { unapprovedTxs } = state.metamask;
+  const unapprovedTxRequests = getApprovalRequestsByType(
+    state,
+    ApprovalType.Transaction,
+  );
+
+  const chainId = getCurrentChainId(state);
+
+  const filteredUnapprovedTxInCurrentNetwork = unapprovedTxRequests.filter(
+    ({ id }) => transactionMatchesNetwork(unapprovedTxs[id], chainId),
+  );
+
+  return filteredUnapprovedTxInCurrentNetwork.length > 0;
+};
+
+// istanbul ignore next: this will be removed after merging https://github.com/MetaMask/metamask-extension/pull/18379
+const hasPendingDecryptApprovals = (state: ApprovalsMetaMaskState) => {
+  // @ts-ignore
+  return state.metamask.unapprovedDecryptMsgCount > 0;
+};
+
+export function hasPendingApprovalsSelector(
+  state: ApprovalsMetaMaskState,
+  approvalType: ApprovalType,
+) {
+  const pendingApprovalRequests = getApprovalRequestsByType(
+    state,
+    approvalType,
+  );
+
   return pendingApprovalRequests.length > 0;
 }
+
+export function hasTransactionPendingApprovalsSelector(
+  state: ApprovalsMetaMaskState,
+) {
+  return (
+    hasUnapprovedTransactionsInCurrentNetwork(state) ||
+    // istanbul ignore next: this will be removed after merging https://github.com/MetaMask/metamask-extension/pull/18379
+    hasPendingDecryptApprovals(state) ||
```

Comment:
This will be changed into `hasPendingApprovalsSelector(state, ApprovalType.EthDecrypt)`only after https://github.com/MetaMask/metamask-extension/pull/18379